### DISCUSSION TITLE
Welcome to bsb_lan Discussions!

### DISCUSSION THREAD

**Discussion by fredlcore** [fredlcore/BSB-LAN#216] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Attention:** For bug reports, please do not write here, but open up a [bug report](https://github.com/fredlcore/BSB-LAN/issues/new?template=bug_report.md).

**Comment by sblasch** [comment]
Hello fredlcore,
i just found your great project, because we got a new brötje heating system BOB 20 B and also we are using Home Assistant already. so i thought it would be a good idea to connect those two systems :-)
Now my actual question is: how can i get an PCB of the adapter?(assembled or not)

greetings from Hamburg
cheers Stephan

**Reply by fredlcore** [reply]
Hi Stephan,

thanks! Yes, BSB-LAN and Home Assistant are working together fine based on feedback by other users (I myself use FHEM) and yes, boards can be ordered from me, just send me an e-mail to "bsb (ät) code-it.de".

Cheers,

Frederik

**Reply by 1coderookie** [reply]
> i just found your great project, because we got a new brötje heating system BOB 20 B and also we are using Home Assistant already. so i thought it would be a good idea to connect those two systems :-)

@sblasch : There is an official integration for HomeAssistant https://www.home-assistant.io/integrations/bsblan/ developed by @liudger - but there's a note that the integration doesn't work with the recent version of BSB-LAN (only up to v1.0). So maybe you'll encounter some problems if you are trying to use the official integration - just wanted to mention that..

**Comment by fredlcore** [comment]
Thanks, @1coderookie, for pointing this out, @liudger, please let us know what has changed and what we can do so that the HomeAssistant integration will continue to work with current versions. That would certainly be a priority. Thanks!

**Comment by AlexanderLux** [comment]
Hi @fredlcore,

thank you for this great projekt. 
Is there a way to buy a complete "BSB-LAN ESP32" Adapter for Olimex ESP32-EVB somewhere? 

Thanks!

**Reply by fredlcore** [reply]
@AlexanderLux: As written on the README page of this project, there are boards available, just send me an e-mail to bsb (ät) code-it.de for further infos.

**Comment by bdevijve** [comment]
Hi @fredlcore !
Many thanks for this great project I'm following since many years now;
I will make the great jump and order pieces to run this with my heating system...
I would prefer RJ45 Lan rather than wifi (with our without POE)

Q1 :
So I guess the ESP32 board could be Olimex ESP32-EVB or Olimex ESP32-POE-ISO, there is not support actually for cheaper WT32-ETH01 boards (available from 5€ on Aliexpress) or even newer ETH01-EVO ?

Can I order Olimex board + BSB-LAN board from you ?

Q2 : I have 2 devices, one with RVS21-85.002.030 board (with AVS55 on top) -> No problem
But do we have any idea about support for " Atlantic ( ref 232510 ) Odysee PI " ? I assume I should open the stuff to check what is the board inside...

Many thanks for this projet !

**Reply by fredlcore** [reply]
Hi @bdevijve, thanks for your kind words and interest in this project!
Regarding your questions: Officially, we only support the boards mentioned on the project site. There have been people using the WT32-ETH01 boards, but you'll have to connect them "on the fly". And if I'm not mistaken, the WT32 doesn't even have an USB connector for power. So that would have to be taken care of as well. To me, the 15 Euro price difference compared to a Olimex POE-ISO would not be worth the hassle.

As for getting the Olimex from me, unfortunately, that's not an option as I would be reliable if something is wrong with that device, that's why I ask people to get them by themselves.

As for the Odysee Pi, I have no knowledge about it - I guess opening it would be the safest way to know about it.

For ordering, as mentioned above, please send me an e-mail to bsb(ät)code-it.de

**Comment by bdevijve** [comment]
Hi,
I notice my BSB port (X86 on RVS21) is already connected to something, I assume I can add wires in parallel to add the BSB-LAN board ?
Many thanks !

**Reply by fredlcore** [reply]
Yes, you can. Just make sure that you use the two rightmost connectors (as shown in the photos in our manual). Sometimes there is a wireless room unit connected to pins 2 and 3 from the right. This combination won't work with BSB-LAN.

**Reply by bdevijve** [reply]
Actually I see 3 wires to X86 : 
![X86](https://github.com/user-attachments/assets/1262a5b3-7e0d-48ce-9e51-26f2dc430049)

Those 3 wires (BSB, M, G+) are connected to a DeltaDore "Typass ATL" box which is the one linked to at least 2 deltadore Tybox 5000 wireless temperature sensors
Can I keep this connected while adding the BSB-LAN board ?

**Reply by fredlcore** [reply]
Yes.

**Reply by bdevijve** [reply]
Many thanks !
One last question before I go : on the AVS55 board I see the X150 that seems equivalent to X86 of RVS21 (BSB,M,etc...); can we connect here ?

**Reply by fredlcore** [reply]
Oh, yes, I didn't realize that before. Do you have a plug for that connector? If yes, could you send me a "bids-eye" photo from above that I could use for my manual, similar to this one:
https://docs.bsb-lan.de/supported_heating_systems.html#RVS21-AVS55
That would be great!

**Reply by bdevijve** [reply]
sure :
![AVS55](https://github.com/user-attachments/assets/24747711-9db9-48d0-8b08-145f41377d0b)

**Reply by fredlcore** [reply]
Thanks a lot!

**Comment by bdevijve** [comment]
Hi Team,
I successfully setup my BSB-LAN (w/ esp32-dev board), compiled the BSB-LAN Software, dumped the data for customization, and then compiled the final BSB-LAN Firmware;

1/ I'm actually looking at details, I'm just wondering why some published data to MQTT are strings and not numbers... like 3113 : "11956 kWh"
which needs us to manipulate the data on homeassistant...

Anyway to get just the number on MQTT ?
Sorry I'm not very familiar with github, and I'm discovering the project so I would not to ask for any dumb change...

2/ It seems I'm not able to successfully change the datetime on my RVS21... If I change directly on the BSB-LAN WebInterface, nothing is changed on my HeatPump UI... some with the NTP sync feature... the HeatPump is not updated...
I would guess I do something wrong...

KR,
Benoît

**Reply by fredlcore** [reply]
Most parameters are transmitted as numbers, but 3113 comes as a string from the heater itself. That is rather unusual, but nothing we can change.
As for the not-updated UI: Do you have an Atlantic Alfea? We just recently figured out that their display unit is different from the other Siemens units. It doesn't query the actual value from the heater before displaying the value (as other display units do), it just displays what it "knows", i.e. the last parameter value that has been set/changed on the display. Again, nothing we can do anything about, but once it's more clear which heaters are affected by this, we'll add info on this in the FAQ...

**Reply by bdevijve** [reply]
Yes, I confirm : I have the Atlantic Alfea, and I guess as you do : this display does show its own clock ;)
Ok, well understood regarding 3113, I'll manage this on the Home Assistant side then !
Many thanks again for your great work !

![image](https://github.com/user-attachments/assets/a85fb4e1-55a9-4e12-9c99-c87d089fadc1)

**Reply by fredlcore** [reply]
I think it's those heating systems that can connect with the wireless room unit without the navipass.io that have this "problem". Then the display thinks it is all-knowing, as everything goes via the display unit, so it always has the most recent parameter values, and therefore, it does not query any parameters when you select them.

**Reply by bdevijve** [reply]
Yes, this unit is using " IO Homecontrol" protocol to speak with some other devices, and probably think he don't need to check data against the RVS21 ...

**Comment by bdevijve** [comment]
Hi !

(just downloaded / build and flashed 4.2.26)
Is there any limit in the number of params we can provide to the "Logging Parameters" field of the webinterface ?

Because I'm actually looking after all changing data inside my RVS21, I identified about 45 parameters to monitor to better understand how this heating is working; while filling this field with all the parameters, I noticed after applying the field is cut somewhere juste after 200 caracters... and the missing parameters are not sent to my MQTT server...

Should I really limit myself ? is it a bug or a limitation ? or is there any workaround for me to monitor those 45 parameters ?

Many thanks again !

**Reply by bdevijve** [reply]
Hi, I see "parameter log_parameters[40]" in the BSB_LAN_config.h file...
I just increased this to to 99, no warning about that ?

**Reply by fredlcore** [reply]
This has been discussed many times over in various places, so I'm not going to address this here again.

**Reply by bdevijve** [reply]
Where are the place for this kind of discussion please ?

**Reply by fredlcore** [reply]
This is discussed in various forums, at least in the FHEM forum, maybe even in some of the closed issues here. In any case, if you decide to change the code, you're up on your own and there is no support from me. This may sound harsh, but there are hundreds of BSB-LAN installations nowadays, and it's great if people improve the code for their own needs, but the impacts/effects can be manifold, so I have to draw a line somewhere in order not to drown in support work that does not benefit the majority of users.

**Comment by bdevijve** [comment]


**Reply by fredlcore** [reply]
If you follow the manual, it clearly states where to file a bug report. Here is not the place.

**Comment by bdevijve** [comment]


**Reply by fredlcore** [reply]
Again, bug reports have to be filed under "Issues", not just add to the general discussion.

**Comment by aa-fr-75** [comment]
Deployment successful using an Olimex ESP32 board. Connected it to an Atlantic VarFree heater without issues. Big thanks to @fredlcore for providing custom parameters—made the setup even smoother! Everything was quite straightforward overall. I'll explore the system further and share more insights later. Great project!

**Comment by sam-4213** [comment]
Hi everyone. My name is Sam and I'm building my Home domotic system with Home assistant. I'm looking to install a new Atlantic Excellia Duo AI 11 - 2024 air-to-water heat pump. It looks like this system is supported by the BSB LAN project. I just wanted to be sure before buying it. Thanks a lot!

**Reply by fredlcore** [reply]
Hi Sam, I'm pretty sure it is compatible, but to be 100% sure, you have to open the heater (you have to do that anyway to connect BSB-LAN later) and compare the controller with the ones we have here, it's most likely an RVS21 or RVS61:
https://docs.bsb-lan.de/supported_heating_systems.html

**Comment by sam-4213** [comment]
Hi. Tks! I ll check. I didnt buy the heater yet.
Sam

**Reply by acarrilloa** [reply]
Hi Sam, I do have it connected to an Atlantic Excellia Duo AI 11 without problems.
Regards,
Alfonso

**Comment by Freeride79** [comment]
Hello everyone,
I have been a new proud owner of a BSB Lan board on an ESP32-Olimex EVB chip for a few days now. I've been trying for some time to make my **Weishaupt WTU oil heating system from 2003** smart and thought for too long that it would work with eBus. 
I can read out my Weishaupt WRSOL2.1 solar thermal system with eBus, but I only found out that the LPB bus of the Weishaupt WTU is something else through the BSB-Lan project. 

Many thanks for the support so far! This is a great project and I was able to read and edit all the parameters of the WTU straight away. 

Since 2021 I have a photovoltaic system with an energy management system in **Openhab** and try to use my oil heating only as a backup. As soon as there is a PV surplus, I heat with air conditioning and in winter also with wood heating connected to the central heating system. That's why I'm interested in connecting as many systems as possible and, at best, controlling them based on forecasts. Since this year, I've also been running **HomeAssistant** and using MQTT to exchange data between the systems. 

**My current question concerns MQTT:** 
BSB-LAN sends many topics to Mosquitto, but **not all of them are sent at the same frequency. You see it on the missing data points** At first the ESP32 ran via WLAN, now I have switched to LAN, but it doesn't change anything. **Often data is only sent again when I actively change something on the web interface**. Have I defined basic settings incorrectly here? Or do I sent too many topics?

Maybe someone can give me some tips. 

Thank you very much!

![grafik](https://github.com/user-attachments/assets/5a35a97b-3090-439c-b594-7d070851f31b)

![grafik](https://github.com/user-attachments/assets/b5905310-a509-4c2b-b98a-66ac36e6eeae)


![grafik](https://github.com/user-attachments/assets/3ecd0869-6c4b-4154-b72b-39a28fdf29a5)

**Reply by fredlcore** [reply]
Hi,
first of all, thanks for your support :)!
As for the interval in which the MQTT parameters are updated, please see here:
https://docs.bsb-lan.de/homeautomation.html#exchanging-data-via-mqtt
> The `/status` topic is updated in four ways:
> - via logging parameters to MQTT as explained above
> - every time a URL query is made to BSB-LAN (can be disabled by **activating** `Only send log parameters to MQTT` in configuration)
> - every time a parameter is changed through the room unit (can be disabled by **activating** `Only send log parameters to MQTT` in configuration)
> - every time the parameter is updated via the `/poll` topic as explained above

There is also a video on the project's YouTube channel which explains how to set up individual update intervals.

**Reply by Freeride79** [reply]
Hi @fredlcore,
the documentation is great, I only had a rough look at MQTT at the beginning because values were sent “automatically”. 
It works better with the appropriate parameters. Now only the defined values are sent and updated regularly. 

Thank you very much.

![grafik](https://github.com/user-attachments/assets/e62d41b3-6e8a-4535-a81f-4a8ec1591a3c)

![grafik](https://github.com/user-attachments/assets/f4f84136-07e5-403b-900e-0128b84eed5c)

![grafik](https://github.com/user-attachments/assets/3c936040-09f8-40fc-8b71-bcc164ee02e5)

**Comment by acarrilloa** [comment]
Hi Fred,
I'm unable to find a way to upload an upgraded version of BSB_LAN.ino.bin via OTA. My web interface looks like this:
![Captura de pantalla 2025-05-27 134647](https://github.com/user-attachments/assets/8894b687-5289-47e0-8ae6-86e20f7fbe0a)
![Captura de pantalla 2025-05-27 134743](https://github.com/user-attachments/assets/3708ebdf-acb6-4d0b-ac84-22cdbe4c85f1)

Any ideas?
Thanks,
Alfonso

**Reply by fredlcore** [reply]
Just follow the instructions in our manual. If OTA is enabled, you should be able to perform the update using the URL mentioned in the instructions.

**Comment by sam4213** [comment]
Hi everyone. I finally installed an atlantic alfea excellia s duo. However the controller does not look like the ones described in supported heating system controllers. Enclosed is a picture of my controller. Can somebody tell me if it is supported or not. Thks a lot!!! Sam
![20251204_224450](https://github.com/user-attachments/assets/be472105-5426-4073-8e75-ce0d43895304)
![20251204_224053](https://github.com/user-attachments/assets/d8dffdcb-01b5-4ddb-a5b8-26a532d1692c)

**Comment by fredlcore** [comment]
That is... interesting. This does not look like a Siemens board, although it says so. It does not look compatible, but could you give me the exact model of your Atlantic? And if possible a link to the (installation) manual? That might clear things up a bit further...

EDIT: I found the POS422 to be a Siemens Climatixic device, Climatixic is Siemens OEM cloud (not really sure what this provides at all). Is it possible that Atlantic uses different controllers in very similar sounding models? When going through the installation manuals that I found just now here:
[https://www.atlantic.be/belgie/technology/warmtepomp](https://www.atlantic.be/belgie/technology/warmtepomp)
Then it seems to me that only the "A.I." models have a compatible Siemens controller, e.g. the "Alfea Extensa A.I. 5 R32" has a compatible controller, but the the "Alfea Extensa S 5" doesn't.
This has never crossed my mind, and I don't know if there are any differences between the "A.I." models and the others, or if Atlantic (just like Brötje) is moving to different controllers with newer models, but is only slightly (if at all) changing the model names.

EDIT 2: Ok, after going through my e-mails, I see that all of the past users who had an Atlantic heat pump had an "A.I." model. Not sure if this means that "A.I." is catering to different requirements, or whether it's replaced with the "S" and other models. The "A.I." models are still listed on the atlantic.fr website, but my French is too weak to check what the (significant) differences are between the Extensa S/M and the Extensa A.I. (and likewise for Excellia). If somebody could clear this up, that would be great.

**Reply by sam4213** [reply]
Thks for the heads up Fred. That s a shame indeed...
Br
Sam

**Reply by sam-4213** [reply]
After reading AlfeaExcellia S12 installation manual it looks like :
- it includes a cozytouch module (no bridge required)
- according to installation manual the network is a Zigbee protocol
I'm trying to find a way not to use cozytouch - I don't want to rely on the cloud for my HA app. I don't know if anyone found a way to control it directly without using cozytouch, if so i'll be pleased to get some support.

**Reply by fredlcore** [reply]
The cozytouch is the room unit, right? I never understood that system well, so I don't know what the bridge is for and where the cozytouch connects to, or if in this case it communicates directly via ZigBee with the heating controller? In that case, it's clear that BSB-LAN can't be used, unfortunately, because it's a wire-based protocol.
